plugins/chat/assets/javascripts/discourse/components/chat-composer.js

```diff
@@ -248,6 +248,38 @@ export default class ChatComposer extends Component {
       return;
     }
 
+    // Check if the message is a reaction to the latest message in the channel.
+    const message = this.draft.message.trim();
+    let reactionCode = "";
+    if (message.startsWith("+")) {
+      const reaction = message.substring(1);
+      // First check if the message is +{emoji}
+      if (replacements[reaction]) {
+        reactionCode = replacements[reaction];
+      } else {
+        // Then check if the message is +:{emoji_code}:
+        const emojiCode = reaction.substring(1, reaction.length - 1);
+        if (emojis.includes(emojiCode)) {
+          reactionCode = emojiCode;
+        }
+      }
+    }
+
+    if (reactionCode) {
+      if (this.lastMessage?.id) {
+        const interactor = new ChatMessageInteractor(
+          getOwner(this),
+          this.lastMessage,
+          this.context
+        );
+
+        await interactor.react(reactionCode, "add");
+      }
+
+      this.resetDraft();
+      return;
+    }
```

Comment:
I do feel like the addition here bloats the `onSend` action quite abit and has an impact on readability. What do you think about moving the logic into another function?